test/cuda.cu

```diff
@@ -0,0 +1,79 @@
+/*******************************************************
+ * Copyright (c) 2020, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <gtest/gtest.h>
+#include <testHelpers.hpp>
+#include <af/array.h>
+#include <af/device.h>
+
+using af::allocV2;
+using af::freeV2;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+TEST(Memory, AfAllocDeviceCUDA) {
```

Comment:
Something like `AllocDevicePointerUsability` instead of `AfAllocDeviceCUDA` is more readable name.

Reply:
its testing af_alloc_device. I think the name is appropriate here.